packages/x-adapter/README.md

```diff
@@ -1,13 +1,462 @@
+them you just have to provide with the `Path` of the data to map, and the `Schema` to apply to it.
+
+###### Types definition
+
+```ts
+// API data models
+interface ApiRequest {
+  q: string;
+}
+interface ApiResponse {
+  users: ApiUser[];
+  total: number;
+}
+interface ApiUser {
+  id: number;
+  email: string;
+  phone: string;
+  address: ApiAddress;
+  company: ApiAddress;
+}
+interface ApiAddress {
+  address: string;
+  city: string;
+  postalCode: string;
+}
+
+// APP data models
+interface AppRequest {
+  query: string;
+}
+interface AppResponse {
+  people: AppUser[];
+  count: number;
+}
+interface AppUser {
+  id: number;
+  contact: {
+    email: string;
+    phone: string;
+    homeAddress: AppAddress;
+    companyAddress: AppAddress;
+  };
+}
+interface AppAddress {
+  displayName: string;
+  postalCode: number;
+  city: string;
+}
+```
+
+###### Schemas and SubSchemas implementation
+
+```ts
+// Address Schema definition
+const addressSchema: Schema<ApiAddress, AppUserAddress> = {
+  displayName: source => `${source.address}, ${source.postalCode} - ${source.city}`,
+  city: 'city',
+  postalCode: source => parseInt(source.postalCode)
+};
+
+// User Schema definition with a subSchema
+const userSchema: Schema<ApiUser, AppUser> = {
+  id: 'id',
+  contact: {
+    email: source => source.email.toLowerCase(),
+    phone: 'phone',
+    homeAddress: {
+      $subSchema: addressSchema,
+      $path: 'address'
+    },
+    companyAddress: {
+      $subSchema: addressSchema,
+      $path: 'address'
+    }
+  }
+};
+```
+
+###### Schema's mapper factory function implementation with subSchemas
+
+```ts
+// Response mapper with user's subSchema implemented
+const responseMapper = schemaMapperFactory<ApiSearchUsersResponse, AppSearchUsersResponse>({
+  people: {
+    $subSchema: userSchema,
+    $path: 'users'
+  },
+  count: 'total'
+});
+
+const requestMapper = schemaMapperFactory<SearchUsersRequest, ApiSearchUsersRequest>({
+  q: 'query'
+});
+
+// Endpoint Adapter Factory function implementation
+export const searchUsersWithContactInfo = endpointAdapterFactory({
+  endpoint: 'https://dummyjson.com/users/search',
+  requestMapper,
+  responseMapper
+});
+```
+
+<br>
+
+#### Using a mutable schema
+
+This feature lets you have some default mappers, and modify or extend them for some concrete
+implementations. To do so, you should use the `createMutableSchema` helper function, and pass as a
+parameter the schema you want to make mutable.
+
+```ts
+// TODO: Creating a mutable schema
+```
+
+Once you have your mutable schema, you can use its available methods to obtain a new schema based on
+it:
+
+- `replace`: Replaces completely the original Schema.
+- `override`: Merges the original schema with the new one.
+- `extends`: Creates a new Schema based on the original one. The original remains unchanged.
+
+```ts
+// TODO: Mutable schema's methods: '$replace', '$override', '$extends'
+```
+
+<br>
+
+### Extend an adapter that uses schemas
```

Comment:
@javieri-empathy @diegopf: This is the part I doubt to add here this way -----> The idea I have is documenting it in `x-adapter-platform` and linking to that readme section when it's done instead of linking to the full package.

Reply:
I would probably leave it here for the next PR. We can link the platform adapter as an example use case. Think that Extending an adapter is a feature provided by `x-adapter`. It is NOT an specific part of `x-platform-adapter`.